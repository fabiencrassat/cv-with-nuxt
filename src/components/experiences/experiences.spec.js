import Experiences from './experiences';
import testUtils from '~~/lib/tests/testUtils';

describe('experiences', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    testUtils.mountProperly(Experiences);
  });

  it('renders properly', () => {
    expect.hasAssertions();
    expect(testUtils.htmlFactory(Experiences)).toMatchInlineSnapshot('""');
  });
  it('renders properly with empty object', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Experiences, {
        propsData: { experiences: {} },
      })
    ).toMatchInlineSnapshot('""');
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Experiences, {
        propsData: {
          experiences: {
            myLastExperience: {
              id: 'myLastExperience',
              jobImage: '/url/to/jobImage1',
              job: 'job1',
              date: 'date1',
              society: {
                siteurl: 'url/to/site',
              },
            },
            myMiddleExperience: {
              id: 'myMiddleExperience',
              jobImage: '/url/to/jobImage2',
              job: 'job2',
              date: 'date2',
              missions: [],
              society: {},
            },
            myFirstExperience: {
              id: 'myFirstExperience',
              jobImage: '/url/to/jobImage3',
              job: 'job3',
              date: 'date3',
              missions: ['mission31', 'mission32'],
            },
          },
        },
      })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });
  it('renders a lot of experiences with show more button', async () => {
    expect.hasAssertions();
    const wrapper = testUtils.factory(Experiences, {
      propsData: {
        experiences: {
          experience1: {
            id: 'experience1',
            jobImage: '/url/to/jobImage',
            job: 'job',
            date: 'date',
          },
          experience2: {
            id: 'experience2',
            jobImage: '/url/to/jobImage',
            job: 'job',
            date: 'date',
          },
          experience3: {
            id: 'experience3',
            jobImage: '/url/to/jobImage',
            job: 'job',
            date: 'date',
          },
          experience4: {
            id: 'experience4',
            jobImage: '/url/to/jobImage',
            job: 'job',
            date: 'date',
          },
          experience5: {
            id: 'experience5',
            jobImage: '/url/to/jobImage',
            job: 'job',
            date: 'date',
          },
          experience6: {
            id: 'experience6',
            jobImage: '/url/to/jobImageAfterClick',
            job: 'jobAfterClick',
            date: 'dateAfterClick',
          },
        },
      },
    });
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.html()).toMatchSnapshot();
    await wrapper.find('button').trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
