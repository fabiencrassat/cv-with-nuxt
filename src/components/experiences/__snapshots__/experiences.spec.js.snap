// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experiences renders properly with props 1`] = `
<div>
  <div id="myLastExperience" class="max-w-sm w-full lg:max-w-full lg:flex mb-4 last:mb-0">
    <div title="job1" class="h-48 lg:h-auto lg:w-64 flex-none bg-cover rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden bg-center" style="background-image: url(/url/to/jobImage1);"></div>
    <div class="border-r border-b border-l border-gray-400 lg:border-l-0 lg:border-t lg:border-gray-400 bg-white rounded-b lg:rounded-b-none lg:rounded-r p-4 flex flex-col justify-between leading-normal lg:w-full">
      <div class="mb-4">
        <p class="text-sm text-gray-700 flex"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" class="fill-current w-5 h-5 mr-2">
            <path fill-rule="evenodd" d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z" clip-rule="evenodd"></path>
          </svg>
          date1
        </p>
        <div class="font-bold text-xl mb-2">
          job1
        </div>
        <!---->
      </div> <a href="url/to/site" target="_blank" rel="noopener noreferrer">
        <societycontent-stub society="[object Object]"></societycontent-stub>
      </a>
    </div>
  </div>
  <div id="myMiddleExperience" class="max-w-sm w-full lg:max-w-full lg:flex mb-4 last:mb-0">
    <div title="job2" class="h-48 lg:h-auto lg:w-64 flex-none bg-cover rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden bg-center" style="background-image: url(/url/to/jobImage2);"></div>
    <div class="border-r border-b border-l border-gray-400 lg:border-l-0 lg:border-t lg:border-gray-400 bg-white rounded-b lg:rounded-b-none lg:rounded-r p-4 flex flex-col justify-between leading-normal lg:w-full">
      <div class="mb-4">
        <p class="text-sm text-gray-700 flex"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" class="fill-current w-5 h-5 mr-2">
            <path fill-rule="evenodd" d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z" clip-rule="evenodd"></path>
          </svg>
          date2
        </p>
        <div class="font-bold text-xl mb-2">
          job2
        </div>
        <!---->
      </div>
      <societycontent-stub society="[object Object]"></societycontent-stub>
    </div>
  </div>
  <div id="myFirstExperience" class="max-w-sm w-full lg:max-w-full lg:flex mb-4 last:mb-0">
    <div title="job3" class="h-48 lg:h-auto lg:w-64 flex-none bg-cover rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden bg-center" style="background-image: url(/url/to/jobImage3);"></div>
    <div class="border-r border-b border-l border-gray-400 lg:border-l-0 lg:border-t lg:border-gray-400 bg-white rounded-b lg:rounded-b-none lg:rounded-r p-4 flex flex-col justify-between leading-normal lg:w-full">
      <div class="mb-4">
        <p class="text-sm text-gray-700 flex"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" class="fill-current w-5 h-5 mr-2">
            <path fill-rule="evenodd" d="M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z" clip-rule="evenodd"></path>
          </svg>
          date3
        </p>
        <div class="font-bold text-xl mb-2">
          job3
        </div>
        <ul class="text-gray-700 text-base">
          <li>
            mission31
          </li>
          <li>
            mission32
          </li>
        </ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
